src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -1603,33 +1603,38 @@ internal static void CheckActionPreference(FunctionContext funcContext, Exceptio
             // set $? to false indicating an error
             context.QuestionMarkVariableValue = false;
 
-            bool anyTrapHandlers = funcContext._traps.Any() && funcContext._traps.Last().Item2 != null;
+            ActionPreference preference = ExceptionHandlingOps.QueryForAction(rte, rte.Message, context);
 
-            if (!anyTrapHandlers && !ExceptionHandlingOps.NeedToQueryForActionPreference(rte, context))
+            // if the exception is new (i.e. it was not rethrown from a naked throw
+            // statement inside of a catch block, and it is not currently being
+            // handled by the system), and if ErrorActionPreference is set to break,
+            // then break immediately into the debugger.
+            if (!rte.WasRethrown &&
+                context.CurrentExceptionBeingHandled == null &&
+                preference == ActionPreference.Break)
```

Comment:
It seems the comment above adds nothing useful - we see directly the same in the code.
I'd remove the dummy comment.

Reply:
I think there is value in clearly identifying what makes an exception "new" to this block of code; however, I can remove it and create an `IsExceptionNew` method to keep that detail in the code itself.

Reply:
Fixed.